### PR TITLE
feat: integrate with signet sdk

### PR DESCRIPTION
**Pull request by chrisling-dev**

### CODE REVIEW

**Comment by tien** [comment]
lgtm, @chrisling-dev There's also `useExtrinsicBatch` which is deprecated & used in one place only. Can you either add signet support to that or just migrate the 1 usage over to `useExtrinsic` instead?

File: apps/web/src/domains/accounts/recoils.ts [hunk 1]
```diff
@@ -127,6 +129,9 @@ export const wagmiAccountsState = atom<Account[]>({
 export const accountsState = selector({
   key: 'Accounts',
   get: ({ get }) => {
+    const signetAccount = get(signetAccountState)
+    if (signetAccount) return [{ ...signetAccount, readonly: false, partOfPortfolio: true }]
```
Comment (tien) [inline apps/web/src/domains/accounts/recoils.ts:133]: Just nit-picking but would be better to do explicit check like `signetAccount !== undefined` if possible.

File: apps/web/src/domains/accounts/recoils.ts [hunk 2]
```diff
@@ -264,3 +272,32 @@ export const AccountWatcher = () => {
 
   return null
 }
+
+export const signetAccountState = atom<InjectedAccount>({
+  key: 'SignetAccountState',
```
Comment (tien) [inline apps/web/src/domains/accounts/recoils.ts:277]: The key here should be `SignetAccount`

File: apps/web/src/domains/accounts/recoils.ts [hunk 3]
```diff
@@ -264,3 +272,32 @@ export const AccountWatcher = () => {
...
+  default: undefined,
+})
+
+export const SignetWatcher = () => {
+  const { inSignet, sdk } = useSignetSdk()
+  const setSignetVaultAccount = useSetRecoilState(signetAccountState)
+
+  useEffect(() => {
+    if (inSignet && sdk) {
```
Comment (tien) [inline apps/web/src/domains/accounts/recoils.ts:286]: Do explicit `undefined` check on `sdk` if possible

File: apps/web/src/domains/common/hooks/useExtrinsic.ts [hunk 4]
```diff
@@ -100,6 +105,12 @@ export function useExtrinsic(
           })
 
           try {
+            if (signingWithSignet && submittable) {
+              const { ok, error, receipt } = await sdk.send(submittable.method.toHex())
+              // this doesnt matter because signet will redirect users away after every transaction
+              if (ok && receipt) reject(new Error('Please ignore this message.'))
+              if (error) reject(new Error(error))
```
Comment (tien) [inline apps/web/src/domains/common/hooks/useExtrinsic.ts:112]: Probably better to do `new Error('Some descriptive msg here', { cause: error })`

**Review by tien — CHANGES_REQUESTED** [review]
Just couple of minor points, also the previous msg about `useExtrinsicBatch`, everything else look good 💪

File: apps/web/src/domains/accounts/recoils.ts (hunk 2 above)
Comment (chrisling-dev) [reply apps/web/src/domains/accounts/recoils.ts]: ahh np, using pattern from signet lol

File: apps/web/src/domains/accounts/recoils.ts (hunk 3 above)
Comment (chrisling-dev) [reply apps/web/src/domains/accounts/recoils.ts]: I removed this check because sdk cannot be undefined in the latest version

**Comment by chrisling-dev** [comment]
> lgtm, @chrisling-dev There's also `useExtrinsicBatch` which is deprecated & used in one place only. Can you either add signet support to that or just migrate the 1 usage over to `useExtrinsic` instead?

Not sure what's the best way to migrate yet given useExtrinsic doesnt return the extrinsic for batching, but instead just exposes the signAndSend function, I assume to migrate you need to build both 'staking.chill' and 'staking.unbond' as extrinsics and then batch them? 

So seems like it needs to be
```tsx
const batchExtrinsic = useExtrinsic('utility', 'batch')
...

return {
  ...
  unbondAll: async (account: string) => {
    // const chillExtrinsic = ... build chill extrinsic
    // const unbondExtrinsic = ...build unbond extrinsic
    return batchExtrinsic.signAndSend(account, {
    //  ... pass in chillExtrinsic and unbondExtrinsic
    })
  }
}
```

Lmk if you have something else in mind or I can migrate according to the snippet above

**Comment by tien** [comment]
@chrisling-dev if you check the function signature of `useExtrinsic`, it also accepts either a submittable or a function that generate a submittable, so you can do something like this:

```ts
const extrinsic = useExtrinsic(useCallback((api) => api.tx.utility.batchAll([
  // ... stuffs goes here
]), []))
```

**Comment by chrisling-dev** [comment]
> @chrisling-dev if you check the function signature of `useExtrinsic`, it also accepts either a submittable or a function that generate a submittable, so you can do something like this:
> 
> ```ts
> const extrinsic = useExtrinsic(useCallback((api) => api.tx.utility.batchAll([
>   // ... stuffs goes here
> ]), []))
> ```

Migrated and removed useExtrinsicBatch! Lmk wyt

File: apps/web/src/domains/common/hooks/useExtrinsic.ts [hunk 5]
```diff
@@ -105,7 +105,7 @@ export function useExtrinsic(
           }
         })()
 
-        const signingWithSignet = signetAccount.address === account
+        const signingWithSignet = signetAccount !== undefined && signetAccount.address === account
```
Comment (tien) [inline apps/web/src/domains/common/hooks/useExtrinsic.ts:108]: Maybe this can be shortened to: `signetAccount?.address === account`

**Review by tien — APPROVED** [review]
lgtm

File: apps/web/src/domains/common/hooks/useExtrinsic.ts (hunk 5 above)
Comment (chrisling-dev) [reply apps/web/src/domains/common/hooks/useExtrinsic.ts]: yeap np, addressed

**Pull request merged**